dealroom_firestore_connector/__init__.py

```diff
@@ -194,6 +201,116 @@ def get_history_doc_refs(
     return tuple(doc.reference for doc in docs)
 
 
+def _validate_dealroomid(dealroom_id: Union[str, int]):
+    if not str(dealroom_id).isnumeric() and dealroom_id != -1:
+        raise ValueError("'dealroom_id' must be an integer")
+
+def _validate_final_url(final_url: str):
+    # Extract method has internally validation rules. Check here:
+    # https://github.com/dealroom/data-urlextract/blob/main/dealroom_urlextract/__init__.py#L33-L35
+    try:
+        extract(final_url)
+    except:
+        raise ValueError("'final_url' must have a url-like format")
+
+def _validate_new_history_doc_payload(payload: dict):
+    """Validate the required fields in the payload when creating a new document"""
+
+    if "final_url" not in payload:
+        raise KeyError("'final_url' must be present payload")
+    
+    _validate_final_url(payload["final_url"])
+    
+    if "dealroom_id" not in payload:
+        raise KeyError("'dealroom_id' must be present payload")
+
+    _validate_dealroomid(payload["dealroom_id"])
+
+def _validate_update_history_doc_payload(payload: dict):
+    """Validate the required fields in the payload when updating"""
+
+    if "final_url" in payload:
+        _validate_final_url(payload["final_url"])
+    
+    if "dealroom_id" in payload:
+        _validate_dealroomid(payload["dealroom_id"])
```

Comment:
another time we can put these in a separate file

Reply:
agree